docs/user_guides/tls_proxy.md

```diff
@@ -419,16 +416,61 @@ openssl x509 -req -days 3650 -in client.csr -out client.pem -CA ca.pem -CAkey ca
+
 ### 3. Run gen.sh to create certificate.
 ```shell
 chmod +x gen.sh
 ./gen.sh
 ```
-They will creat 9 files about ca.key, ca.pem, ca.srl, server.key, server.pem, server.csr, client.key, client.pem, client.csr.
 
-## Modify config
+Finally, you will get 9 files, including ca.key, ca.pem, ca.srl, server.key, server.pem, server.csr, client.key, client.pem, client.csr. If you want to know the ```gen.sh``` file detail, read the next chapter.
+
+### 4. Get the detail of generating certificate files
+
+To implement the ssl/tls mutual authentication, it mainly involves three parts, including ca, client, server. The ca is to ensure that the certificate used by the client and server to communicate with each other is legal.
+
+> Tips: If you want to know more details of using the openssl command, execute ```man openssl``` or visit [the openssl manual page](https://www.openssl.org/docs/)
+
+- Generate an RSA private key for the ca. If you want to know more details of this command, execute ```openssl genras -help``` or ```man openssl```.
+
+```shell
+openssl genpkey -algorithm RSA -out ca.key
+```
+
+- Make a certificate request and generate a ca certificate. This process needs some basic information of ca and you can get a ```ca.pem``` file which is a ca certificate used to generate client and server certificates. The ```x509``` option outputs a certificate instead of a certificate request, which means self-signing.
+
+```shell
+openssl req -new -x509 -key ca.key -out ca.pem -days 3650 -subj "/C=$Country/ST=$State/L=$Location/O=$Organization/OU=$Organizational/CN=$CommonName"
+```
+
+Generating a server certificate is same as generating a client certificate, including three steps. The following is an example of a server to illustrate this process.
+
+- Generate a server private key, you will get a ```server.key``` file.
+
+```shell
+openssl genpkey -algorithm RSA -out server.key
+```
+
+- Generate a certificate signing request file, and some basic server info is required. When the command complete to execute, you will get a ```server.csr``` file.
+
+```shell
+openssl req -new -nodes -key server.key -out server.csr -days 3650 -subj "/C=$Country/O=$Organization/OU=$Organizational/CN=$CommonName" -config ./openssl.cnf -extensions v3_req
+```
+
+- Sign the certificate. You need to enter the ```server.csr``` file, the ```ca.key``` file and the ```ca.pem``` file. The meaning of the ```CAcreateserial``` command option is that the CA serial number file is created if it does not exist, and this is why you get a ```ca.srl``` file.
```

Comment:
- Open the `server.csr, the `ca.key` and the `ca.pem` files to sign the certificate. The `CAcreateserial` command option is used to create a CA serial number file if it does not exist. You will get a `ca.srl` file after choosing this command option.